src/View/InstallQuickInput.ts

```diff
@@ -87,17 +93,57 @@ export async function showInstallQuickInput() {
     const versions =
         toolchains.map((value) => value.info.version !== undefined ? value.info.version.str() : '');
     const versionGroups: vscode.QuickPickItem[] = versions.map((label) => ({label}));
-    state.version = await input.showQuickPick({
+    const version = await input.showQuickPick({
       title,
       step: 3,
       totalSteps: 3,
       placeholder: 'Pick toolchain version',
       items: versionGroups,
+      buttons: [updateButton],
       shouldResume: shouldResume
     });
+    if (version instanceof InnerButton) {
+      console.log('press the refresh button');
+      return (input: MultiStepInput) => updateBackend(input, state);
```

Comment:
It's kinda difficult to read this code to me. Could you explain a bit?
- `if (version instanceof InnerButton) {`
  - Is this case covers when `InnerButton` was pressed? If so, maybe other variable name instead of `version` would be better to read this code.
- L107 returns a function but code after L110 does not return function. I wonder if this is OK.

Reply:
> Is this case covers when InnerButton was pressed? If so, maybe other variable name instead of version would be better to read this code.

This `showQuickPick` function returns promise and it returns the InnerButton if user clicks the refresh button and the `version` QuickPickItem if user clicks the version. I don't know how to name this variable. The L74 is in a similar situation and it uses `type` as name because it returns `type` QuickPickItem. What is a better name than version? Please give me an idea. :)

> L107 returns a function but code after L110 does not return function. I wonder if this is OK.

Yes, it's okay. Because this step is the last step for this installation.
When user selects the version number, this code jumps to L155 to install a new version.

The reason why the L107 returns the function is to show the version list again after updated apt cache. To show this multi step input UI in next scenario, it must need the state and this variable so I forwarded it.